File: PS/Javascript/implementation/Baekjoon_1110.js
const fs = require("fs");

const filePath =
  process.platform === "linux" ? "/dev/stdin" : __dirname + "/data.txt";

const number = Number(fs.readFileSync(filePath).toString().trim());

let answer = 0;

let cycleNumber = number;

while (true) {
  const newNumber = Number.parseInt(cycleNumber / 10) + (cycleNumber % 10);

  cycleNumber = (cycleNumber % 10) * 10 + (newNumber % 10);

  answer += 1;

  if (number === cycleNumber) {
    break;
  }
}

console.log(answer);

File: PS/Python/implementation/Baekjoon_1110.py
import sys

if __name__ == "__main__":
  N = int(sys.stdin.readline().strip())
  new_number = N
  answer = 0

  while True:
    if new_number < 10:
      new_number = int(str(new_number) + str(new_number))

    elif new_number >= 10:
      string_number = str(new_number)
      
      sum_number = int(string_number[0]) + int(string_number[1])

      if sum_number < 10:
        new_number = int(string_number[1] + str(sum_number))
      
      elif sum_number >= 10:
        new_number = int(string_number[1] + str(sum_number)[1])

    answer += 1

    if new_number == N:
      break
  
  print(answer)
  